react/features/polls/components/PollChoice.js

```diff
@@ -0,0 +1,188 @@
+
+    /**
+     * Can be removed, or changed. Need remove and text handlers.
+     */
+    editable: boolean,
+
+    /**
+     * True if item can't be voted.
+     */
+    disabled: boolean,
+
+    /**
+     * Wether this item is selected before by the user or not.
+     */
+    selected: boolean,
+
+    /**
+     * Text of the voting option.
+     */
+    text: string,
+
+    /**
+     * Container text update method.
+     */
+    textChangeHandler?: Function,
+
+    /**
+     * Option voting handling method.
+     */
+    voteHandler?: Function,
+
+    /**
+     * Number of votes for this option.
+     */
+    votes: number
+
+};
+
+/**
+ * Poll option view component.
+ */
+class PollChoice extends Component<Props, *> {
+
+    /**
+     * Constructor.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onClickHandler = this._onClickHandler.bind(this);
+        this._onTextChange = this._onTextChange.bind(this);
+        this._onVoteChange = this._onVoteChange.bind(this);
+    }
+
+    /**
+     * Component render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled, editable, selected } = this.props;
+        const itemControls = this.props.editable
+            ? (
+                <div
+                    className = { 'pollChoiceRemoveButtonContainer' } >
+                    <button
+                        id = { 'pollChoiceRemoveButton' }
+                        onClick = { this._onClickHandler }
+                        type = { 'button' } >
+                        ×
+                    </button>
+                </div>
+            ) : (
+                <div
+                    className = { 'pollChoiceVotesContainer' } >
+                    <label>
+                        { this.props.votes }
+                    </label>
+                </div>
+            );
+
+        return (
+            <li>
+                <div
+                    className = { 'pollChoiceContainer' } >
+                    <div
+                        className = { 'pollVoteButtonContainer' } >
+                        <input
+                            checked = { selected }
+                            disabled = { disabled }
+                            id = { 'pollVoteButton' }
+                            onClick = { this._onVoteChange }
+                            type = { 'radio' }
+                            value = { '' } />
+                    </div>
+                    <div
+                        className = { 'pollChoiceTextContainer' } >
+                        <FieldTextStateless
+                            disabled = { !editable }
+                            id = { 'pollChoiceText' }
+                            isLabelHidden = { true }
+                            onChange = { this._onTextChange }
+                            shouldFitContainer = { true }
+                            value = { this.props.text } />
```

Comment:
This poll choice seems to be used for both the poll editing view and the voting view. However, with the voting view hovering over each choice causes the disabled cursor to appear, which may be a confusing experience. I'm tempted to say this may be a case where it's just worthwhile to make two separate components; one to display editing of a choice within an input and another to display the choice with no input or any editing-related UI.